app/assets/v2/js/cart-data.js

```diff
@@ -246,6 +247,74 @@ class CartData {
     applyCartMenuStyles();
   }
 
+  // Updates localStorage with the latest grant data from the backend and returns the new cart
+  static async updateCart() {
+    // Read cart data from local storage
+    const cartData = this.loadCart();
+
+    // Make sure no grants have empty currencies, and if so default to 0.001 ETH. This prevents cart
+    // from being stuck as 'loading' if a currency is empty
+    cartData.forEach((grant, index) => {
+      if (!grant.grant_donation_currency) {
+        cartData[index].grant_donation_currency = 'ETH';
+        cartData[index].grant_donation_amount = '0.001';
+      }
+    });
+
+    // Get list of all grant IDs and fetch updated grant data from backend
+    const grantIds = cartData.map((grant) => grant.grant_id);
+    const url = `${window.location.origin}/grants/v1/api/grants?pks=${grantIds.join(',')}`;
```

Comment:
If you check the diff, you'll notice we were already querying this same endpoint, and we just moved where it happens. So I don't think this will slow anything down compared to the previous baseline?

If so that means this is just an optimization then, and I guess my question is do we want to make optimizations on this PR? If so, that's ok, just want to check

Reply:
is not the same point not even the same prop names, please check the endpoints on your local with a grant id and you will see the diff just in one grant on local (means without the huge metadata related grants and wall of love)  I get a diff of 1.2kb vs 2.7kb so yes will slow a lot.